### PR TITLE
Impl m2 test deploy script

### PR DESCRIPTION
**Pull request by gpsanant**
Implementation of M2 deploy script for AVS team.

This is just the M1 script with the BLSPubkeyCompendium added and default config unpausing everything.

@samlaf does this work for you?

### CODE REVIEW

**Review by samlaf — APPROVED** [review]
LGTM. I'm not that familiar with the previous deploy scripts so they all look roughly the same to me.

When we say "everything is unpaused", that means literally everything? Since we don't have slashing or payments etc for M2, shouldnt not everything be unpaused?

File: script/testing/M2_Deploy_From_Scratch.s.sol [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+                    new TransparentUpgradeableProxy(
+                        address(baseStrategyImplementation),
+                        address(eigenLayerProxyAdmin),
+                        abi.encodeWithSelector(StrategyBaseTVLLimits.initialize.selector, strategyConfigs[i].maxPerDeposit, strategyConfigs[i].maxDeposits, IERC20(strategyConfigs[i].tokenAddress), eigenLayerPauserReg)
+                    )
+                ))
+            );
+        }
+
+        blsPublicKeyCompendium = new BLSPublicKeyCompendium();
```
[the remark this answers is not in the document]
Comment (gpsanant) [reply script/testing/M2_Deploy_From_Scratch.s.sol:257]: doesn't need to be one?

File: script/testing/M2_Deploy_From_Scratch.s.sol [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+            deployed_strategies, strategyConfigs[strategyConfigs.length - 1].tokenSymbol,
+            address(deployedStrategyArray[strategyConfigs.length - 1])
+        );
+
+        string memory deployed_addresses = "addresses";
+        vm.serializeAddress(deployed_addresses, "eigenLayerProxyAdmin", address(eigenLayerProxyAdmin));
+        vm.serializeAddress(deployed_addresses, "eigenLayerPauserReg", address(eigenLayerPauserReg));
+        vm.serializeAddress(deployed_addresses, "slasher", address(slasher));
+        vm.serializeAddress(deployed_addresses, "slasherImplementation", address(slasherImplementation));
+        vm.serializeAddress(deployed_addresses, "delegation", address(delegation));
```
[the remark this answers is not in the document]
Comment (gpsanant) [reply script/testing/M2_Deploy_From_Scratch.s.sol:304]: too late my g. slasher should've been called slashingManager

[the remark this answers is not in the document]
Comment (samlaf) [reply script/testing/M2_Deploy_From_Scratch.s.sol:304]: That too, but that's a separate issue. But the DelegationManager is actually called delegationManager, but it's output in the deployment json file as `delegation` only. That's what I was referring to.

**Pull request merged**